src/runtime/plan/planning-result.ts

```diff
@@ -151,7 +153,80 @@ export class PlanningResult {
     return true;
   }
 
-  append({suggestions, lastUpdated = new Date(), generations = []}) {
+  merge({suggestions, lastUpdated = new Date(), generations = [], contextual = true}, arc: Arc): boolean {
```

Comment:
consider creating a type for the options object that is used by set/append/merge

```
 type PlanningResultOptions = {
   suggestions: Suggestion[];
   lastUpdated?: Date;
   generations: ??;
   contextual?: boolean;
}
```

then add that type to the call points.

Reply:
done